vpr/src/place/place.cpp

```diff
@@ -1312,8 +1339,21 @@ static e_move_result try_swap(const t_annealing_state* state,
         rlim = state->rlim;
     }
 
-    //Generate a new move (perturbation) used to explore the space of possible placements
-    e_create_move create_move_outcome = move_generator.propose_move(blocks_affected, move_type, rlim, placer_opts, criticalities);
+    e_create_move create_move_outcome;
+
+    //When manual move toggle button is active, the manual move window asks the user for input.
+    if (manual_move_flag) {
+#ifndef NO_GRAPHICS
+        draw_manual_moves_window("");
```

Comment:
suggest one helper function for this (calls the four lines of code) to try to shorten this long routine.
create_move_outcome = mm_display_and_propose (...)